modules/taboolaBidAdapter.js

```diff
@@ -151,12 +151,61 @@ export const spec = {
     if (!serverResponse || !serverResponse.body) {
       return [];
     }
-
+    const bids = [];
+    const fledgeAuctionConfigs = [];
     if (!serverResponse.body.seatbid || !serverResponse.body.seatbid.length || !serverResponse.body.seatbid[0].bid || !serverResponse.body.seatbid[0].bid.length) {
-      return [];
+      if (!serverResponse.body.ext || !serverResponse.body.ext.igbid || !serverResponse.body.ext.igbid.length) {
+        return [];
+      }
+    } else {
+      bids.push(...converter.fromORTB({response: serverResponse.body, request: request.data}).bids);
+    }
+    if (isArray(serverResponse.body.ext?.igbid)) {
+      serverResponse.body.ext.igbid.forEach((igbid) => {
+        let buyerdata;
+        try {
+          buyerdata = JSON.parse(igbid.igbuyer[0]?.buyerdata)
```

Comment:
you can use [`safeJSONParse`](https://github.com/prebid/Prebid.js/blob/a0683559f87f96584c2f9aaeb556caa9ac269066/src/utils.js#L1010-L1014) to reduce the boilerplate. 

I am not sure about the purpose of doing it here as well as in the loop below: if the first buyer has invalid `buyerdata` the whole `igbid` is discarded. But if the second one does, only the buyer is discarded. Is that intended?

Reply:
Hi @dgirardi,
thanks for the suggestion. Changed the parse to safeParse.
Regarding the second part of comment - you got correct the intention. I want to discard only the buyer in case of failure inside of the loop and not the full action config.

Reply:
Hi @dgirardi,
Is there any other issues or PR can be merged?